modules/sendtx/src/lib.rs

```diff
@@ -0,0 +1,28 @@
+use futures::stream::Stream;
+use futures::StreamExt;
+use solana_lite_rpc_core::block_store::BlockInformation;
+use tokio::pin;
+use tokio::task::JoinHandle;
+use tokio_stream::wrappers::errors::BroadcastStreamRecvError;
+
+pub async fn start_module(
+    block_info_steam: impl Stream<Item = BlockInformation> + std::marker::Send + 'static,
+    slot_steam: impl Stream<Item = Result<u64, BroadcastStreamRecvError>> + std::marker::Send + 'static,
+) -> JoinHandle<std::io::Result<()>> {
+    let handle = tokio::spawn(async move {
+        pin!(slot_steam);
```

Comment:
It's the main loop that process input stream data. The select wait for data from the 2 stream. When a data is available, it is processed. The process should be very short or spawn in a specific task. This pattern can be used to avoid to use mutex of thread safe share structure as you can process share structure in one task and use an adapted algorithm because all data access are done in this loop.. 
If you're interested this is a more complex (and very raw) implementation that manage geyser account,-slot-epoch-block, bootstrap, RPC access and leader schedule and dump processing: https://github.com/blockworks-foundation/solana-rpc-v2/blob/0efb26c01b6c02013a74bf6aa72072564a95dd5a/stake_aggregate/src/main.rs#L206
I've to refactor everything to make is more readable, but the idea is here. As you can see, there's no lock or mutex to share the data, and I'm always sure that no task can block another. For example to manage the processing of all stake account (500000) that can take more than 1s and still allow processing incoming stake account from geyser during the leader schedule calculus, I split the account storage, one part contain all account to calculate the schedule and the other part cache the incoming one that are integrated after the processing. During the leader schedule process, the notifications are not needed. They are only needed for the next epoch.